contracts/Vault.vy

```diff
@@ -1793,9 +1812,12 @@ def sweep(token: address, amount: uint256 = MAX_UINT256):
     """
     assert msg.sender == self.governance
     # Can't be used to steal what this Vault is protecting
-    assert token != self.token.address
     value: uint256 = amount
     if value == MAX_UINT256:
         value = ERC20(token).balanceOf(self)
+
+    if token == self.token.address:
+        value = self.token.balanceOf(self) - self.totalIdle
```

Comment:
Just to be clear if balanceOF is < totalIdle this is safeMath so it will revert, correct?

Which IMO is ok if this is ever an edge case